### PR TITLE
E2E VRF: validate multi hop via vrf

### PR DESCRIPTION
**Pull request by fedepaol**
Extending the set of containers with another two reacheabile via multihop through the interface in the vrf.

The multihop setup function is made generic so that can be applied to the containers in vrf too.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -323,8 +323,14 @@ jobs:
           sed -i 's/quay.io\/metallb\/controller:main/quay.io\/metallb\/controller:dev-amd64/g' bin/metallb-operator.yaml
           sed -i 's/native/frr/g' bin/metallb-operator.yaml
           kubectl apply -f bin/metallb-operator.yaml
-          sleep 30s
 
+      - name: Ensure MetalLB operator is ready
+        run: |
+          sleep 5
+          while [ "$(kubectl get pods -n metallb-system -l control-plane='controller-manager' -o jsonpath='{.items[*].status.containerStatuses[0].ready}')" != "true" ]; do
```
Comment (oribon) [inline .github/workflows/ci.yaml:330]: can you add some retry counter? probably won't trigger but better be safe

File: e2etest/bgptests/infra_setup.go [hunk 2]
```diff
@@ -177,26 +228,26 @@ func InfraTearDown(cs *clientset.Clientset, containers []*frrcontainer.FRR) erro
 
 // multiHopSetUp connects the ebgp-single-hop container to the multi-hop-net network,
```
Comment (oribon) [inline e2etest/bgptests/infra_setup.go:229]: can you change this comment + update the docs (can be in a different pr I guess)?

File: e2etest/bgptests/infra_setup.go [hunk 3]
```diff
@@ -147,27 +164,61 @@ func VRFContainersSetup(cs *clientset.Clientset) ([]*frrcontainer.FRR, error) {
...
+
+func infraTearDown(cs *clientset.Clientset, containers []*frrcontainer.FRR, nextHop nextHopSettings, filter func(*frrcontainer.FRR) bool) error {
+	multiHopRoutes, err := container.Networks(nextHop.nextHopContainerName)
+	if err != nil {
+		return err
+	}
+
+	filtered := make([]*frrcontainer.FRR, 0)
+	for _, c := range containers {
+		if !filter(c) {
```
Comment (oribon) [inline e2etest/bgptests/infra_setup.go:212]: I think it's more clear doing this the other way (if it matches the filter it is considered)

File: e2etest/bgptests/infra_setup.go [hunk 4]
```diff
@@ -18,22 +18,39 @@ import (
...
+	kindNetwork         = "kind"
+	vrfNetwork          = "vrf-net"
+	vrfName             = "red"
+	defaultRoutingTable = ""
+	metalLBASN          = 64512
+	metalLBASNVRF       = 64513
+	externalASN         = 4200000000
 )
 
+type nextHopSettings struct {
```
Comment (oribon) [inline e2etest/bgptests/infra_setup.go:30]: neat

File: e2etest/bgptests/infra_setup.go (hunk 3 above)
Comment (fedepaol) [reply e2etest/bgptests/infra_setup.go:212]: fair enough, but then I'll change the name because filter doesn't sound right.

**Review by oribon — APPROVED** [review]
lgtm
